staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/admission_test.go

```diff
@@ -276,7 +279,67 @@ func setupTestCommon(t *testing.T, compiler ValidatorCompiler) (plugin admission
 		// wait for informer factory to shutdown
 		fakeInformerFactory.Shutdown()
 	})
-	return handler, dynamicClient.Tracker(), fakeClient.Tracker(), handler.evaluator.(*celAdmissionController)
+
+	// Wait for admission controller to begin its object watches
+	// This is because there is a very rare (0.05% on my machine) race doing the
+	// initial List+Watch if an object is added after the list, but before the
+	// watch it could be missed.
+	//
+	// This is only due to the fact that NewSimpleClientset above ignores
+	// LastSyncResourceVersion on watch calls, so do it does not provide "catch up"
+	// which may have been added since the call to list.
+	if !cache.WaitForNamedCacheSync("initial sync", testContext.Done(), handler.evaluator.HasSynced) {
+		t.Fatal("failed to do perform initial cache sync")
+	}
+
+	// WaitForCacheSync only tells us the list was performed.
+	// Keep changing an object until it is observable, then remove it
+
+	i := 0
+
+	dummyPolicy := &v1alpha1.ValidatingAdmissionPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "dummypolicy.example.com",
+			ResourceVersion: fmt.Sprint(i),
+		},
+	}
+
+	dummyBinding := &v1alpha1.ValidatingAdmissionPolicyBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "dummybinding.example.com",
+			ResourceVersion: fmt.Sprint(i),
+		},
+	}
+
+	require.NoError(t, fakeClient.Tracker().Create(definitionsGVR, dummyPolicy, dummyPolicy.Namespace))
+	require.NoError(t, fakeClient.Tracker().Create(bindingsGVR, dummyBinding, dummyBinding.Namespace))
+
+	wait.PollWithContext(testContext, 100*time.Millisecond, 300*time.Millisecond, func(ctx context.Context) (done bool, err error) {
+		defer func() {
+			i += 1
+		}()
+
+		dummyPolicy.ResourceVersion = fmt.Sprint(i)
+		dummyBinding.ResourceVersion = fmt.Sprint(i)
+
+		require.NoError(t, fakeClient.Tracker().Update(definitionsGVR, dummyPolicy, dummyPolicy.Namespace))
+		require.NoError(t, fakeClient.Tracker().Update(bindingsGVR, dummyBinding, dummyBinding.Namespace))
+
+		if obj, err := controller.getCurrentObject(dummyPolicy); obj == nil || err != nil {
```

Comment:
Should we also check the resource version here?

Reply:
is this resolved with new update that removes RV?

Reply:
You don't want to test that the obj retrieved is the object that was just modified?

Reply:
This is just an existence test. Objects were added to the test tracker between the informer's List+Watch sequence. 

Since the fake client set does not respect the `LastSyncedResourceVersion` in ListOptions, any objects added by the test between the list but before the watch are dropped.

This PR attempts to work around this by:

1. calling WaitForCacheSync to assure the initial List has been performed. 
2. show that the controller can receive events through its watch by showing it knows of the existence of an object created after it's initial list call.
3. now proceeding with the test

Maybe there is a cleaner way to wait for the watch. This seems to work and am open to suggestions

Reply:
This is probably okay for now, but I would think maybe modifying the fake client set to respect `LastSyncedResourceVersion` would be better, then we wouldn't need to work around this.

Reply:
Agreed. This pr ensures the test doesn't flake in the meantime